integration-tests/test_e2e.py

```diff
@@ -0,0 +1,36 @@
+"""
+This Python module contains integration tests for rhc.
+It uses pytest-client-tools Python module.More information about this
+module could be found: https://github.com/ptoscano/pytest-client-tools/
+"""
+
+import contextlib
+import pytest
+from time import sleep
+from utils import (
+    yggdrasil_service_is_active,
+    prepare_args_for_connect,
+)
+
+
+@pytest.mark.parametrize("auth", ["activation-key", "basic"])
+def test_rhc_client_connect_e2e(rhc, test_config, auth, external_inventory, subman):
+    """
+    Test RHC Client connect e2e using basic auth and activation key
+    test_steps:
+        1. Run RHC client connect
+    expected_results:
+        1. The rhcd service is running and active
+        2. A new host is created in Inventory
+        3. The host rhc client id matches the rhc client id collected from the machine
+    """
+    with contextlib.suppress(Exception):
+        rhc.disconnect()
+    command_args = prepare_args_for_connect(test_config, auth=auth)
+    command = ["connect"] + command_args
+    rhc.run(*command)
+    assert rhc.is_registered
+    assert yggdrasil_service_is_active()
+    sleep(30)  # small wait time for inventory sync up
```

Comment:
There is 'loop_until' method in conftest.py. I guess it could help with this statement. It is necessary to define predicate for a loop to stop only. https://github.com/RedHatInsights/insights-client/blob/master/integration-tests/conftest.py#L43  an example of using the function: https://github.com/RedHatInsights/insights-client/blob/master/integration-tests/test_display_name_option.py#L71